### ISSUE TITLE
OSSEC Windows Agent fails to apply XPath filters in XML.

### ISSUE THREAD

**Issue by gjahchan** [wazuh/wazuh#100]
Example:

![screenshot - 2017-04-11 07_13_57](https://cloud.githubusercontent.com/assets/22189754/24892619/ad3e7930-1e86-11e7-8289-64fcd37cc347.png)

Results in zero security events being forwarded by the agent. With the specified filter, the OSSEC agent should forward all security events except those generated by the Windows Firewall.

The OSSEC documentation says otherwise:
![screenshot - 2017-04-11 07_23_01](https://cloud.githubusercontent.com/assets/22189754/24892790/cf486512-1e87-11e7-8e9f-3af4c8d6319b.png)

**Comment by vikman90** [comment]
Hi @gjahchan,

as you know, you can define a query using both XPath or XML. The thing is that the agent configuration itself is XML and it produces an issue when the configuration reader parses the query (and interprets the query content).

You should escape the `<` symbols in order to avoid them from being interpreted by the Wazuh agent.

You can achieve the goal that you propose by using this XPath query:

```
<localfile>
  <location>Security</location>
  <log_format>eventchannel</log_format>
  <query>Event/System[EventID \< 5150 and EventID > 5159]</query>
</localfile>
```

You can also use the same XML query that you propose if you escape the `<` symbols (and add a `"` after `Security`):

```
<localfile>
  <location>Security</location>
  <log_format>eventchannel</log_format>
  <query>
    \<QueryList>
      \<Query Id="0"\ Path="Security">
        \<Select Path="Security">*\</Select>
        \<Suppress Path="Security">*[System[(EventID >= 5150 and EventID &lt;= 5159)]]\</Suppress>
      \</Query>
    \</QueryList>
  </query>
</localfile>
```

I hope this work for you. Please try to use one of these queries and write back to us with your results.

Best regards.

**Comment by gjahchan** [comment]
Tried both suggestions. The agent restarts  after adding either one of the suggested filters, but _stops forwarding security events_ after adding it. Other logs continue working normally.

As soon as I remove the filters and restart the agent, the agent starts forwarding the security log.

**Comment by gjahchan** [comment]
Tested WEC. Some messages go through fine, some other are chopped when picked up by the agent.

Following is an example of a chopped 4663 event: (from agent debug log):

2017/05/03 08:51:46 ossec-agent: DEBUG: Sending message to server: '2017 May 03 07:00:04 WinEvtLog: Security: AUDIT_SUCCESS(4663): Microsoft-Windows-Security-Auditing: (no user): no domain: Hostname: (no message)'

When comparing the same event in Security and in ForwardedEvents Logs there is one differences in the XML detailed view, and two differences when selecting Copy -> Copy as Text.

The first line reads as follows (in detailed XML view and Copy ->Copy Text):
In security log:
\<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">

In Forwarded Log:
\<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event" xml:lang="en-US">

In Event Viewer detailed XML view, the last three lines are identical in both cases:
  \<Data Name="ResourceAttributes" /> 
 \</EventData\>
\</Event\>

However, if I select "Copy -> Copy details as text" from Event Viewer:

From Security Log (Different from View):
    \<Data Name="ResourceAttributes">
    \</Data>
  \</EventData>
\</Event>

From ForwardedEvents log (same as view):
    \<Data Name="ResourceAttributes" />
  \</EventData>
\</Event>

Could this anomaly (is it a bug in WEC?) in the last \<Data Name in ForwardedEvents Log be wreaking havoc with 4663 event parsing by the agent?

This is an example. I have not performed sufficient testing to determine which event IDs cause problems.

**Comment by nicolaspapp** [comment]
Hello @gjahchan, first of all I want to apologize for all the time this issue has been without an answer. I have been recently looking at this issue, and I have notice it seems to be part of a known issue of Eventchannel around version 3.8 and that was solved for release 3.11, here is one of the issues that referenced this problem: #2713 

I have been making a few checks in order to make sure that is problem is not persisting. I tested with a few Event IDs in Wazuh v3.13.1 and I see both the `<query>` filter working and the event reaching the manager without any truncation in its log.

If you could please see if this problem is fixed after upgrading your Wazuh version I would kindly appreciate it.

**Comment by wazuhci** [comment]
## Footprint metrics information
|                                  |                                            |
|---------------------------------|--------------------------------------------|
| **Main release candidate issue #** |                                          |
| **Main footprint metrics issue #** |                                          |
| **Version** | 4.3.0                                      |
| **Release candidate #** | RC1                                         |
| **Tag** | https://github.com/wazuh/wazuh/tree/4.3.0-rc1 |

## Stress test documentation
### Packages used
- Repository: `packages-dev.wazuh.com`
- Package path: `pre-release`
- Package revision: `1`

- **Jenkins build**: https://devel.ci.wazuh.info/job/Test_stress/1643/

---

<details><summary>Manager</summary>

+ <details><summary>Plots</summary>

    ![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_manager_centos/plots/monitor-manager-Test_stress_B1643_manager-pre-release_CPU.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_manager_centos/plots/monitor-manager-Test_stress_B1643_manager-pre-release_Disk.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_manager_centos/plots/monitor-manager-Test_stress_B1643_manager-pre-release_Disk_Read.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_manager_centos/plots/monitor-manager-Test_stress_B1643_manager-pre-release_Disk_Written.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_manager_centos/plots/monitor-manager-Test_stress_B1643_manager-pre-release_FD.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_manager_centos/plots/monitor-manager-Test_stress_B1643_manager-pre-release_PSS.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_manager_centos/plots/monitor-manager-Test_stress_B1643_manager-pre-release_Read_Ops.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_manager_centos/plots/monitor-manager-Test_stress_B1643_manager-pre-release_RSS_MAXMIN.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_manager_centos/plots/monitor-manager-Test_stress_B1643_manager-pre-release_RSS.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_manager_centos/plots/monitor-manager-Test_stress_B1643_manager-pre-release_SWAP.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_manager_centos/plots/monitor-manager-Test_stress_B1643_manager-pre-release_USS.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_manager_centos/plots/monitor-manager-Test_stress_B1643_manager-pre-release_VMS.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_manager_centos/plots/monitor-manager-Test_stress_B1643_manager-pre-release_Write_Ops.png)



    </details>
+ <details><summary>Logs and configuration</summary>

    [ossec_Test_stress_B1643_manager_2021-12-24.zip](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_manager_centos/logs/ossec_Test_stress_B1643_manager_2021-12-24.zip)



    </details>
+ <details><summary>CSV</summary>

    [monitor-manager-Test_stress_B1643_manager-pre-release.csv](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_manager_centos/data/monitor-manager-Test_stress_B1643_manager-pre-release.csv)



    </details>

</details>

<details><summary>Centos agent</summary>

+ <details><summary>Plots</summary>

    ![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/plots/monitor-agent-Test_stress_B1643_centos-pre-release_CPU.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/plots/monitor-agent-Test_stress_B1643_centos-pre-release_Disk.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/plots/monitor-agent-Test_stress_B1643_centos-pre-release_Disk_Read.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/plots/monitor-agent-Test_stress_B1643_centos-pre-release_Disk_Written.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/plots/monitor-agent-Test_stress_B1643_centos-pre-release_FD.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/plots/monitor-agent-Test_stress_B1643_centos-pre-release_PSS.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/plots/monitor-agent-Test_stress_B1643_centos-pre-release_Read_Ops.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/plots/monitor-agent-Test_stress_B1643_centos-pre-release_RSS_MAXMIN.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/plots/monitor-agent-Test_stress_B1643_centos-pre-release_RSS.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/plots/monitor-agent-Test_stress_B1643_centos-pre-release_SWAP.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/plots/monitor-agent-Test_stress_B1643_centos-pre-release_USS.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/plots/monitor-agent-Test_stress_B1643_centos-pre-release_VMS.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/plots/monitor-agent-Test_stress_B1643_centos-pre-release_Write_Ops.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/plots/Test_stress_B1643_centos_agentd_state_AgentD_Number_of_events_buffered.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/plots/Test_stress_B1643_centos_agentd_state_AgentD_Number_of_generated_events.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/plots/Test_stress_B1643_centos_agentd_state_AgentD_Number_of_messages.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/plots/Test_stress_B1643_centos_agentd_state_AgentD_Status.png)



    </details>
+ <details><summary>Logs and configuration</summary>

    [ossec_Test_stress_B1643_centos_2021-12-24.zip](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/logs/ossec_Test_stress_B1643_centos_2021-12-24.zip)



    </details>
+ <details><summary>CSV</summary>

    [monitor-agent-Test_stress_B1643_centos-pre-release.csv](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/data/monitor-agent-Test_stress_B1643_centos-pre-release.csv)
[Test_stress_B1643_centos_agentd_state.csv](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_centos/data/Test_stress_B1643_centos_agentd_state.csv)



    </details>

</details>

<details><summary>Ubuntu agent</summary>

+ <details><summary>Plots</summary>

    ![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/plots/monitor-agent-Test_stress_B1643_ubuntu-pre-release_CPU.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/plots/monitor-agent-Test_stress_B1643_ubuntu-pre-release_Disk.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/plots/monitor-agent-Test_stress_B1643_ubuntu-pre-release_Disk_Read.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/plots/monitor-agent-Test_stress_B1643_ubuntu-pre-release_Disk_Written.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/plots/monitor-agent-Test_stress_B1643_ubuntu-pre-release_FD.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/plots/monitor-agent-Test_stress_B1643_ubuntu-pre-release_PSS.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/plots/monitor-agent-Test_stress_B1643_ubuntu-pre-release_Read_Ops.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/plots/monitor-agent-Test_stress_B1643_ubuntu-pre-release_RSS_MAXMIN.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/plots/monitor-agent-Test_stress_B1643_ubuntu-pre-release_RSS.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/plots/monitor-agent-Test_stress_B1643_ubuntu-pre-release_SWAP.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/plots/monitor-agent-Test_stress_B1643_ubuntu-pre-release_USS.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/plots/monitor-agent-Test_stress_B1643_ubuntu-pre-release_VMS.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/plots/monitor-agent-Test_stress_B1643_ubuntu-pre-release_Write_Ops.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/plots/Test_stress_B1643_ubuntu_agentd_state_AgentD_Number_of_events_buffered.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/plots/Test_stress_B1643_ubuntu_agentd_state_AgentD_Number_of_generated_events.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/plots/Test_stress_B1643_ubuntu_agentd_state_AgentD_Number_of_messages.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/plots/Test_stress_B1643_ubuntu_agentd_state_AgentD_Status.png)



    </details>
+ <details><summary>Logs and configuration</summary>

    [ossec_Test_stress_B1643_ubuntu_2021-12-24.zip](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/logs/ossec_Test_stress_B1643_ubuntu_2021-12-24.zip)



    </details>
+ <details><summary>CSV</summary>

    [monitor-agent-Test_stress_B1643_ubuntu-pre-release.csv](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/data/monitor-agent-Test_stress_B1643_ubuntu-pre-release.csv)
[Test_stress_B1643_ubuntu_agentd_state.csv](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_ubuntu/data/Test_stress_B1643_ubuntu_agentd_state.csv)



    </details>

</details>

<details><summary>Windows agent</summary>

+ <details><summary>Plots</summary>

    ![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_windows/plots/monitor-winagent-Test_stress_B1643_windows-pre-release_CPU.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_windows/plots/monitor-winagent-Test_stress_B1643_windows-pre-release_Disk.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_windows/plots/monitor-winagent-Test_stress_B1643_windows-pre-release_Disk_Read.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_windows/plots/monitor-winagent-Test_stress_B1643_windows-pre-release_Disk_Written.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_windows/plots/monitor-winagent-Test_stress_B1643_windows-pre-release_Handles.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_windows/plots/monitor-winagent-Test_stress_B1643_windows-pre-release_Read_Ops.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_windows/plots/monitor-winagent-Test_stress_B1643_windows-pre-release_RSS_MAXMIN.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_windows/plots/monitor-winagent-Test_stress_B1643_windows-pre-release_RSS.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_windows/plots/monitor-winagent-Test_stress_B1643_windows-pre-release_USS.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_windows/plots/monitor-winagent-Test_stress_B1643_windows-pre-release_VMS.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_windows/plots/monitor-winagent-Test_stress_B1643_windows-pre-release_Write_Ops.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_windows/plots/Test_stress_B1643_windows_agentd_state_AgentD_Number_of_events_buffered.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_windows/plots/Test_stress_B1643_windows_agentd_state_AgentD_Number_of_generated_events.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_windows/plots/Test_stress_B1643_windows_agentd_state_AgentD_Number_of_messages.png)
![](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_windows/plots/Test_stress_B1643_windows_agentd_state_AgentD_Status.png)



    </details>
+ <details><summary>Logs and configuration</summary>

    [ossec_Test_stress_B1643_windows_2021-12-24.zip](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_windows/logs/ossec_Test_stress_B1643_windows_2021-12-24.zip)



    </details>
+ <details><summary>CSV</summary>

    [monitor-winagent-Test_stress_B1643_windows-pre-release.csv](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_windows/data/monitor-winagent-Test_stress_B1643_windows-pre-release.csv)
[Test_stress_B1643_windows_agentd_state.csv](https://ci.wazuh.com/data/Test_stress/pre-release/4.3.0/B1643-5m/B1643_agent_windows/data/Test_stress_B1643_windows_agentd_state.csv)



    </details>

</details>

<details><summary>macOS agent</summary>

+ <details><summary>Plots</summary>

    MACOS_PLOTS


    </details>
+ <details><summary>Logs and configuration</summary>

    MACOS_LOGS


    </details>
+ <details><summary>CSV</summary>

    MACOS_CSV


    </details>

</details>

<details><summary>Solaris agent</summary>

+ <details><summary>Plots</summary>

    SOLARIS_PLOTS


    </details>
+ <details><summary>Logs and configuration</summary>

    SOLARIS_LOGS


    </details>
+ <details><summary>CSV</summary>

    SOLARIS_CSV


    </details>

</details>
